sql/src/main/java/io/crate/memory/OnHeapMemoryManager.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.memory;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+
+import java.util.function.IntConsumer;
+
+/**
+ * A MemoryManager that allocates buffers on the heap.
+ * It doesn't account the used memory directly, but indirectly via an injected `IntConsumer`.
+ * This IntConsumer is also responsible to de-account the used-bytes eventually.
+ */
+public class OnHeapMemoryManager implements MemoryManager {
+
+    private final IntConsumer accountBytes;
```

Comment:
Is it ok to expect an `IntConsumer` here? `RamAccounting.addBytes()` consumes a long and `Int.MAX_VALUE` could be exhausted on executions which consumes a lot of memory...

Reply:
Ah but it can never be more than an `Int` cause its the `capacity`. Got it, ignore, sorry ;-)

Reply:
Yes, `allocate(int capacity)`  is limited to int range, and `Unpooled.buffer` supports only int.

`RamAccounting` internally can still grow larger than the int range, it is just that a single allocation can't exceed Integer.MAX_VALUE.